File: bank-service/src/main/java/rs/edu/raf/service/impl/CountryServiceImpl.java
package rs.edu.raf.service.impl;

import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;
import rs.edu.raf.dto.CountryCreateDTO;
import rs.edu.raf.dto.CountryDTO;
import rs.edu.raf.exceptions.CountryNotFoundException;
import rs.edu.raf.mapper.CountryMapper;
import rs.edu.raf.model.Country;
import rs.edu.raf.repository.CountryRepository;
import rs.edu.raf.service.CountryService;

import java.util.List;

@Service
@AllArgsConstructor
public class CountryServiceImpl implements CountryService {
    private CountryRepository countryRepository;
    private CountryMapper countryMapper;

    @Override
    public CountryDTO createCountry(CountryCreateDTO countryCreateDTO) {
        return countryMapper.countryToCountryDTO(countryRepository.save(countryMapper.countryCreateDTOtoCountry(countryCreateDTO)));
    }

    @Override
    public List<CountryDTO> getAllCountries() {
        return countryRepository.findAll().stream().map(countryMapper::countryToCountryDTO).toList();
    }

    @Override
    public void deleteCountry(Long id) {
        Country country = countryRepository.findById(id)
                .orElseThrow(()->new CountryNotFoundException(id));

        countryRepository.delete(country);
    }
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/service/impl/CountryServiceImpl.kt
package org.example.bankservicekotlin.service.impl

import org.example.bankservicekotlin.dto.CountryCreateDTO
import org.example.bankservicekotlin.dto.CountryDTO
import org.example.bankservicekotlin.exception.CountryNotFoundException
import org.example.bankservicekotlin.mapper.CountryMapper
import org.example.bankservicekotlin.model.Country
import org.example.bankservicekotlin.repository.CountryRepository
import org.example.bankservicekotlin.service.CountryService
import org.springframework.stereotype.Service

@Service
class CountryServiceImpl(
    private val countryRepository: CountryRepository,
    private val countryMapper: CountryMapper): CountryService {
    override fun createCountry(countryCreateDTO: CountryCreateDTO): CountryDTO {
        return countryMapper.countryToCountryDTO(
            countryRepository.save(
                countryMapper.countryCreateDTOtoCountry(
                    countryCreateDTO
                )
            )
        )
    }

    override fun getAllCountries(): List<CountryDTO> {
        return countryRepository.findAll().stream().map(countryMapper::countryToCountryDTO).toList()
    }

    override fun deleteCountry(id: Long) {
        val country: Country = countryRepository.findById(id)
            .orElseThrow { CountryNotFoundException(id) }

        countryRepository.delete(country)
    }
}